lib/plug/router/utils.ex

```diff
@@ -148,6 +148,12 @@ defmodule Plug.Router.Utils do
       [] ->
         build_path_clause(rest, params, [segment | match], guards, post_match, context, compiled)
 
+      [{prefix_size, match_length}] when match_length == 2 ->
+        suffix_size = byte_size(segment) - prefix_size - 2
+        <<prefix::binary-size(prefix_size), "\\:", suffix::binary-size(suffix_size)>> = segment
+        escaped_segment = [prefix <> ":" <> suffix | match]
+        build_path_clause(rest, params, match, guards, post_match, context, compiled)
```

Comment:
We should probably allow "*" to be escaped too, which means you can rewrite this code to:

```suggestion
        <<prefix::binary-size(prefix_size), ?\\, char, suffix::binary-size(suffix_size)>> = segment
        escaped_segment = [prefix <> <<char>> <> suffix | match]
        build_path_clause(rest, params, match, guards, post_match, context, compiled)
```

Can you please add this change too?

Reply:
Yes, you need to escape the escape character!